tests/upgrade/conftest.py

```
@@ -0,0 +1,54 @@
+    if not repo_url:
+        raise ValueError("upgrade_repo_url must be provided via --tc=upgrade_repo_url:<url>")
+    if not repo_ref:
+        raise ValueError("upgrade_repo_ref must be provided via --tc=upgrade_repo_ref:<ref>")
+    if not script_relative_path:
+        raise ValueError("upgrade_script_path must be provided via --tc=upgrade_script_path:<relative-path>")
+
+    base_dir = tmp_path_factory.mktemp("mtv-autodeploy")
+    clone_dir = base_dir / "repo"
+
+    LOGGER.info(f"Cloning {repo_url} (ref={repo_ref}) into {clone_dir}")
+    subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)